packages/SidePanel/src/SidePanel.js

```diff
@@ -0,0 +1,190 @@
+  kind: PropTypes.oneOf(["default", "child"]),
+
+  /** Disable scroll overflow on document.body when SidePanel is open. */
+  disableBodyOverflow: PropTypes.bool,
+};
+
+const defaultProps = {
+  // this is an internal prop use by <SidePanel.Group /> (Group.js)
+  isInline: false, // eslint-disable-line
+  kind: "default",
+  offsetY: 0,
+  onAfterClose: () => {},
+  onClose: null,
+  onAfterOpen: () => {},
+  width: "33%",
+  zIndex: 100,
+  disableBodyOverflow: true,
+};
+
+function SidePanel(props) {
+  // Props
+  const {
+    onAfterClose,
+    onAfterOpen,
+    onClose,
+    width,
+    isInline, // eslint-disable-line
+    kind,
+    offsetY,
+    isOpen,
+    ...moreProps
+  } = props;
+
+  // Hooks
+  const [isSidePanelVisible, setIsSidePanelVisible] = React.useState(props.isOpen);
+  const offsetScroll = useOffsetScroll(offsetY);
+  useBodyOverflow(props.disableBodyOverflow);
+  useEscapeKey(props.isOpen, onClose);
+
+  // Refs
+  const refTrigger = React.useRef(null);
+  const refSidePanelContent = React.useRef(null);
+  const refHeader = React.useRef(null);
+
+  // Extracts
+  const {
+    "SidePanel.FocusTrap": focusTrapExtracted,
+    "SidePanel.Footer": footerExtracted,
+    "SidePanel.Header": headerExtracted,
+    "SidePanel.Overlay": overlayExtracted,
+    "SidePanel.Trigger": triggerExtracted,
+    children,
+  } = extractChildren(props.children, [
+    "SidePanel.FocusTrap",
+    "SidePanel.Footer",
+    "SidePanel.Header",
+    "SidePanel.Overlay",
+    "SidePanel.Trigger",
+  ]);
+
+  const handleAnimationEnd = () => {
+    if (!props.isOpen) {
+      setIsSidePanelVisible(false);
+      onAfterClose();
+
+      if (triggerExtracted) {
+        refTrigger.current.focus();
+      }
+      return;
+    }
+
+    onAfterOpen();
+  };
+
+  React.useEffect(() => {
+    if (props.isOpen) {
+      setIsSidePanelVisible(true);
+    }
+  }, [props.isOpen]);
+
+  const extendedFocusTrapOptions = focusTrapExtracted ? focusTrapExtracted.props : {};
+  const fallbackFocus = () => {
+    return refHeader.current || refSidePanelContent.current;
+  };
+
+  const focusTrapOptions = {
+    fallbackFocus,
+    clickOutsideDeactivates: true,
+    ...extendedFocusTrapOptions,
+  };
+
+  let sidePanel = null;
+
+  if (isSidePanelVisible) {
+    const dialog = (
+      <Dialog
+        onAnimationEnd={handleAnimationEnd}
+        header={headerExtracted}
+        footer={footerExtracted}
+        onClose={onClose}
+        refSidePanelContent={refSidePanelContent}
+        width={width}
+        kind={kind}
+        isInline={isInline}
+        refHeader={refHeader}
+        offsetY={offsetScroll}
+        isOpen={isOpen}
+        {...moreProps}
+      >
+        {children}
+      </Dialog>
+    );
+
+    if (isInline) {
+      sidePanel = dialog;
+    } else {
+      sidePanel = ReactDOM.createPortal(
+        <React.Fragment>
+          <FocusTrap focusTrapOptions={focusTrapOptions}>
+            <div>{dialog}</div>
```

Comment:
Does it really need the `<div>`?

Reply:
yes it does, the FocusTrap needs to have only 1 element as children can't have two neither a `<Fragment />`